sync-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -73,25 +90,50 @@ public static SQLDatabase createSQLDatabase(String dbFilename) throws IOExceptio
      *         can not be created
      */
     public static SQLDatabase openSqlDatabase(String dbFilename) throws IOException {
+        return openSqlDatabase(dbFilename, new NullKeyProvider());
+    }
+
+    /**
+     * SQLCipher-based implementation for opening database.
+     * @param dbFilename full file path of the db file
+     * @param provider Key provider object storing the SQLCipher key
+     *                 Supply a NullKeyProvider to use a non-encrypted database.
+     * @return {@code SQLDatabase} for the given filename
+     * @throws IOException if the file does not exists, and also
+     *         can not be created
+     */
+    public static SQLDatabase openSqlDatabase(String dbFilename, KeyProvider provider) throws IOException {
```

Comment:
Our external code is using a combination of open/create and I'd like to review that separately from this PR.